### ISSUE TITLE
Forum Hierarchy / Posting

### ISSUE THREAD

**Issue by wtl420** [TheMinusWorld/Mayhem-Whereabouts#5]
We need to figure out how we want forums to be structured. Posting falls under this because ultimately the structure is to bring the maximum amount of Posting.

The traditional layout is:

- Board Index
  - Categories
    - Forums
       - Forums
       - Threads
         - Posts
  - Members
    - Personal Messages

I think we should go for a structure more like:

- Board Index
  - Latest Threads
  - Categories
    - Categories
    - Threads
      - Posts
- Accounts
  - Threads
  - Characters
    - Threads

Basically replace personal messages with private threads, and get rid of the concept of category / forum- they should be functionally equivalent. We might want to think about tags here as well.

Blocked by #4

**Comment by DarkoJohnny** [comment]
I agree with the concept of having the latest (3-5) posted in threads be displayed at the top of the main board page. I personally don't use such systems but it seems others do and it would help draw further activity in when it's already striking, making for a positive effect.

I have to say that I don't want Conversations gone as they're a great MW feature and very useful for back and forths during KGs. I don't think Private Threads would work as well. We also need to keep the Account stuff up at the top in the corner since what Character you have active and your Conversations are key for some of the main things people use the Forums for. Namely, again, KGs. The current MW UI for account controls needs work (and far less hover menus) but I feel like that can be fixed up rather than thrown out, as it definitely works well. Something I'd like to see is the current account box stretched out across the very top of the forum with quick access to all profile customization/Character swapping features as their own buttons with no menu hopping.

As for the forum layout, yeah, we can axe some of the fluff and simply have a list of the forum categories all in one box. We don't have that many to begin with anyways. I'd kind of like it if users could also rearrange the forums themselves to their own pleasing. So if someone is here for KG they could move the forum for that to the top for themselves and if they don't want to touch the Arts and Crafts forum they could move it down to the bottom.

**Comment by keithpickering** [comment]
Instead of latest threads it'd be cool if we had some kind of algorithm for thread popularity. I wouldn't be all that interested in a list of 3-5 brand new threads, but I would be interested in a list of 3-5 threads that a lot of people have been posting in within the last few hours. The name we discussed previously for this category was **Fresh Beef** and I still think that's funny.

I want little avatars all over so I can see where people are. Forums have always had that little "users browsing this forum" or "users browsing this thread" and I can think of countless times when that little list has kept me in a forum/thread I would've otherwise left. People want to be where the activity is. The problem is you can't tell where people are until you actually enter the thread. Every forum and thread in their respective lists should display tiny avatars of users who are currently viewing them. If I click one of these avatars at any time, I should get a little popover with some quick info about that user and a link to their profile. 

Another random idea to drive activity: make it known when a thread has a bunch of spells. Maybe each thread should have a Magic ranking ((number of spells / total posts) * 100) and we can display that somehow.

**Comment by DarkoJohnny** [comment]
Ok, how about we have something like a selection of 2-3 of the most popular threads at the current time at the top, and then the latest made threads (1-2 of them) right underneath that?

Love the showing user thread activity on the thread index a lot and will pull people into threads rather than simply keep them in them. MW's current system of having avatars in a lot of places to show activity is one of its best aspects and building on that is a great idea.

**Comment by wtl420** [comment]
I think we could make it some kind of user preference what shows up there. I think something more curated than most recent makes the most sense for guests but most recent might make the most sense for active posters. We should also have a way to filter out threads we don't care to see or maybe even entire boards.

**Comment by wtl420** [comment]
I don't really like the idea of conversations or traditional personal messages because I think they're basically the same thing as posting but arbitrarily remove features. We could definitely include different kinds of layout options for viewing posts and some could resemble conversations as we currently have them.

**Comment by FunWithDespair** [comment]
I like PMs entirely because they're accessible through a different menu and as such, are unlikely to lead to any confusion. This being said, I do hate how PMs remove features like say, spells. For stuff like KG, I think a good Autoplay for example would be great for account PMs and other such things. I think there's definitely a compromise that works between a private forum and our current conversations that essentially just allows for full site feature access.

**Comment by keithpickering** [comment]
What if PMs were basically just private forum threads you could invite
certain users to? They can use the exact same components and functionality
as regular threads otherwise

**Comment by keithpickering** [comment]
I like private threads, this whole interface is JS based so we can pop new messages in without a refresh either way. Maybe threads could have either an Expanded or a Compact view, regular threads could default to expanded and private threads could default to compact

**Comment by FunWithDespair** [comment]
We're assuming Private Threads like what wtl wrote with your Private Threads listed separately from your Public Threads in an easily viewable forum, right? I don't have much issue with that system entirely because it's basically the exact same thing as having a PM section except you get full topic features.

If Private Threads were to replace PMs though, I think that this would literally have to be the only way to do it . In the case of events like KG and certain garrisons, things would become immensely cluttered for the host if Private Threads could be made anywhere.

**Comment by wtl420** [comment]
Sorry for the confusion, I don’t think private threads should go in the
same place that public threads do. I imagine for private threads they’d be
attached to accounts or characters but you’d also be able to apply labels
to them to sort them for your own personal needs.

**Comment by wtl420** [comment]
Also I think Anonymous posting should be treated as a first class feature
and probably allowed anywhere. I’m thinking even accounts without
characters selected may be able to anonymously post.

**Comment by DarkoJohnny** [comment]
Anonymous posting as a primary feature instead of tacked on would be great. Maybe leave a log whenever an anonymous post's connected account is reviewed to make sure that other site staff can know if someone else is abusing their access to such information when it isn't necessary. Linking them directly to accounts is fine since the reason behind doing so is solely for moderation purposes.

**Comment by wtl420** [comment]
Adding a log for unmasking anonymous posters is a really smart idea. Definitely want to include that in the mod tools. I want to link them to accounts for moderation purposes but making it awkward / tracked if there's any unmasking would give people a lot more confidence in the anonymous feature.

**Comment by FunWithDespair** [comment]
I definitely agree with the unmask log
